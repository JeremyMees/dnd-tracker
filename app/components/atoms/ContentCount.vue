<script setup lang="ts">
defineProps<{
  count: number | undefined
  max: number
  loading?: boolean
}>()
</script>

<template>
  <ClientOnly>
    <UiSkeleton
      v-if="loading || !isDefined(count)"
      class="w-10 h-4"
    />
    <span
      v-else-if="isDefined(count)"
      class="text-[12px] text-muted-foreground"
      :class="{ '!text-destructive': count >= max }"
    >
      {{ count }}/{{ max }}
    </span>

    <template #fallback>
      <UiSkeleton class="w-10 h-4" />
    </template>
  </ClientOnly>
</template>
