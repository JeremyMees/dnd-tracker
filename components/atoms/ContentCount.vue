<script setup lang="ts">
defineProps<{
  count?: number
  max: number
  loading?: boolean
}>()
</script>

<template>
  <UiSkeleton
    v-if="loading || !count"
    class="w-10 h-4"
  />
  <span
    v-else-if="count"
    class="text-[12px] text-muted-foreground"
    :class="{ '!text-destructive': count >= max }"
  >
    {{ count }}/{{ max }}
  </span>
</template>
