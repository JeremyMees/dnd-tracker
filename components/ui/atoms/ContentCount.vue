<script setup lang="ts">
defineProps<{ count: number, max: number }>()
</script>

<template>
  <span
    class="text-[12px] text-slate-300"
    :class="{ '!text-danger': count >= max }"
  >
    {{ count }}/{{ max }}
  </span>
</template>
