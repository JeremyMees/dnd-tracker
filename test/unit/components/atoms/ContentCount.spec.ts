import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import ContentCount from '~/components/atoms/ContentCount.vue'

interface Props {
  count: number | undefined
  max: number
  loading?: boolean
}

const props: Props = {
  count: 1,
  max: 2,
}

describe('ContentCount', async () => {
  it('Should match snapshot', async () => {
    const component = await mountSuspended(ContentCount, { props })
    expect(component.html()).toMatchSnapshot()
  })

  it('Should render with default props correctly', async () => {
    const component = await mountSuspended(ContentCount, { props })

    expect(component.find('[data-test-count]').text()).toBe(`${props.count}/${props.max}`)
  })

  it('Should render a zero count', async () => {
    const component = await mountSuspended(ContentCount, { props: { ...props, count: 0 } })

    expect(component.find('[data-test-count]').text()).toBe(`0/${props.max}`)
  })

  it('Should render a loading state', async () => {
    const component = await mountSuspended(ContentCount, { props: { ...props, loading: true } })

    expect(component.find('[data-test-skeleton]').exists()).toBeTruthy()

    component.setProps({ loading: false, count: undefined })
    await nextTick()

    expect(component.find('[data-test-skeleton]').exists()).toBeTruthy()
  })

  it('Should render red text when count is greater or equal to max', async () => {
    const component = await mountSuspended(ContentCount, { props: { ...props, count: 2 } })

    expect(component.find('[data-test-count]').attributes('class')).toContain('text-destructive')
  })
})
