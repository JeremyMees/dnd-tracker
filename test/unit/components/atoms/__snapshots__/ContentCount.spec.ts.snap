// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentCount > Should match snapshot 1`] = `"<span data-test-count="" class="text-[12px] text-muted-foreground">1/2</span>"`;
